softcut-lib/src/Voice.cpp

```diff
@@ -144,6 +153,12 @@ void Voice::setRecFlag(bool val) {
 	}
     }
     recFlag = val;
+    if (!val) {
+	// turn off rec once if active
+        if (sch.getRecOnceActive()) {
+	    setLoopFlag(false);
```

Comment:
@schollz  i did have a question about this line. why should disabling writes also change the loop flag status if rec-once is active?

Reply:
i missed this. it does seem like this functionality shouldn't impact loop status.

Reply:
this was a route to allow disabling the rec-once behavior. if you call rec-once and then you decide you don't want the loop to occur, the only way to switch it off is to call `rec(<voice>,0)` which will reset it via `setLoopFlag(false)`.

the activity check through `getRecOnceActive` is sorta silly now that I think about it - that function should be true if rec-once is flagged to be active *or* active. I thought it was an optimization so you don't have to reset the rec-once flag everytime `rec(<voice>,0)` comes around (checking vs setting) but its sort-of silly now that I look at it again.

it could just be changed to

```cpp
    if (!val) {
	// reset rec-once behavior
	setLoopFlag(false);
    }
```

and the `getRecOnceActive()` function could be totally eliminated.

on another note - if this behavior is unwanted - perhaps there can be a separate plumbing function like you mentioned `rec_once_cancel`. I think the canceling behavior itself is useful. I'm writing a script now using `rec_once` and I think its pretty likely I will need something to cancel.

I'm happy to update another PR iterating on any of these things

Reply:
oh s***, nvm, hat's a bug. it should be `setRecOnceFlag` not `setRecFlag`!